auth0/nextjs-auth0
# We should cleanup unused cookies when switching between chunked and unchunked

### Description

If the size of the session grows from <4096 Bytes to >4096 Bytes, the middleware will create new `appSession.0` ..., `appSession.n` chunked cookies.

But it will not remove the orginal single `appSession`, and since the cookie-store [prefers](https://github.com/auth0/nextjs-auth0/blob/v1.0.0/src/auth0-session/cookie-store.ts#L75-L79) `appSession`, it will continue to read the stale single cookie rather than the updated chunked cookies.

### References

fixes #289 

### Testing

- Login with small session (eg. database user with no custom claims)
- Observe that you have a single `appSession` cookie
- Hit some middleware that adds a large value to your session (making the session > 4096 Bytes)
- Observe that you have new `appSession.0` and `appSession.1` cookies - you should not have the `appSession` cookie
- You should be able to read the value of the new property on subsequent requests


### Checklist
